integrationtests/online_dataset/test_online_dataset.py

```diff
@@ -438,7 +441,11 @@ def test_dataset() -> None:
 def main() -> None:
     try:
         test_dataset()
+    except Exception as e:
+        print(f"An exception occured: {e}")
+        raise
```

Comment:
@MaxiBoether just want to ensure it's intended: why do we want to re-raise the exception here? Or is this a leftover from debugging?
If we do intend to re-raise exception, we can simply do
```python
test_dataset()
cleanup_storage_database()
cleanup_dataset_dir()
```
, or if we need the printed message, we can do 
```python
try:
    test_dataset()
    print(...)
else:
    print(...)
    cleanup_storage_database()
    cleanup_dataset_dir()
```

But anyway, we should clean up things no matter the tests go wrong/successful right?

Reply:
My idea here was:

If there is any exception, I want to log and then fail such that we have an exit code -1. in both cases, I want to cleanup at the end. The reason is that i had the feeling at some point the test was exciting with -1 without printing any exception stacktrace, this gives me a custom string to search for in the logs. I dont fully understand yet whether the way i did this is problematic or not

Reply:
Sorry I think I misunderstood how `try` block works. It's not problematic at all!